config/main.py

```diff
@@ -726,6 +729,47 @@ def reload():
     else:
         click.secho('Buffer definition template not found at {}'.format(buffer_template_file), fg='yellow')
 
+#
+# 'snmp-community' command
+#
+@config.command('snmp-community')
+@click.argument('community-name', metavar='<community_name>', required=True)
+def snmp_community(community_name):
+    """Add SNMP community string"""
+
+    """Serialization of a Python dict from a SNMP YAML File """
+    with open(SNMP_COMMUNITY_FILE, 'r') as stream:
+        try:
+            snmp_community_dict = yaml.safe_load(stream)
+        except yaml.YAMLError as error:
+            click.echo("SNMP config file not loaded correctly with error {}".format(error))
+            raise
+
+    if 'snmp_rocommunity' in snmp_community_dict.keys():
+        existing_community_name = snmp_community_dict['snmp_rocommunity']
+        if 'snmp_rocommunities' not in snmp_community_dict.keys():
+            snmp_community_dict['snmp_rocommunities'] = [existing_community_name]
+        snmp_community_dict['snmp_rocommunities'].append(community_name)
+    else:
+        snmp_community_dict['snmp_rocommunity'] = community_name
```

Comment:
>community_name [](start = 50, length = 14)

Suggest also append to snmp_rocommunities.

Reply:
appended _**community_name**_ to snmp_rocommunities.

Reply:
I guess you need to check whether `snmp_rocommunities` exists or not, as above if-block.

---
In reply to: [340822493](https://github.com/Azure/sonic-utilities/pull/687#discussion_r340822493) [](ancestors = 340822493)